src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,26 @@
+const { ApiError } = require('../exceptions/ApiError.js');
+const { jwtService } = require('../services/jwt.service.js');
+
+function authMiddleware(req, res, next) {
+  const authHeader = req.headers['authorization'];
+
+  if (!authHeader) {
+    throw ApiError.Unauthorized();
+  }
+
+  const [, accessToken] = authHeader.split(' ');
+
+  if (!accessToken) {
+    throw ApiError.Unauthorized();
+  }
+
+  const userData = jwtService.validateAccessToken(accessToken);
+
+  if (!userData) {
+    throw ApiError.Unauthorized();
+  }
+
+  next();
```

Comment:
After validating the access token, it's a common practice to attach the `userData` to the `req` object (e.g., `req.user = userData;`). This allows subsequent middleware and route handlers to access the authenticated user's data. Consider adding this step before calling `next()`.